mautic/mautic
# M3: Refactor AbTestHelper in all bundles

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? |Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8067
| BC breaks? | AbTestHelper class removed, use DetermineWinnerSubscriber instead
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Refactor AbTestHelper in all bundles to get rid of static methods and use events instead. The goal is to get rid of MauticFactory.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create A/B tests for Email and Page. Verify, that all winner criteria work like before and winner is being chosen (seems we can't use submissions for emails, because we can't use forms in emails. Also, in bounce rate criteria for landing pages, we are choosing the highest rate as a winner instead of the lowest one, maybe even the rate calculation is not right. But that is not a part of this task).
